codecs/jxl/enc/jxl_enc.cpp

```diff
@@ -0,0 +1,57 @@
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+#include "jxl/enc_file.h"
+
+using namespace emscripten;
+
+thread_local const val Uint8Array = val::global("Uint8Array");
+
+struct JXLOptions {
+  // 1 = slowest
+  // 7 = fastest
+  int speed;
+};
+
+val encode(std::string image, int width, int height, JXLOptions options) {
+  jxl::CompressParams cparams;
+  jxl::PassesEncoderState passes_enc_state;
+  jxl::CodecInOut io;
+  jxl::PaddedBytes bytes;
+  jxl::ImageBundle* main = &io.Main();
+
+  cparams.speed_tier = static_cast<jxl::SpeedTier>(options.speed);
+  cparams.color_transform = jxl::ColorTransform::kNone;
```

Comment:
Better to leave color_transform at its default.

Reply:
Setting it to kNone is very much untested and will likely completely break things in lossy mode – this is only intended for lossless encoding, where you don't want to do irreversible color transforms. The default is kXYB which is what the lossy encoder is expecting to work with. The only reason we even allow kNone (and kYCbCr) in lossy mode is that we need it for lossless JPEG recompression.

Reply:
Ah, so that seems to fix the assertion error at speed=2. However, now the colors are wrong. I guess I need to specify sRGB somewhere?
![Screenshot 2020-08-28 at 12 18 15](https://user-images.githubusercontent.com/234957/91555355-c7e72780-e928-11ea-9df5-64a68817aaba.png)

Reply:
Actually, no. I use `SetFromSRGB()`, so it’s already marked as sRGB input. I think that’s why I originally ended up using `kNone` because it fixed the colors 😅

Reply:
After adding alpha support just now, I wonder if the discolorization is a _decoder_ issue? I am doing this clamping on an `Image3F` which might be wrong? 

https://github.com/GoogleChromeLabs/squoosh/blob/321996807e7caefb8d72316d40d6e830fb2dc3c8/codecs/jxl/dec/jxl_dec.cpp#L51-L61

Reply:
Digging into the same issue here. [This issue](https://gitlab.com/wg1/jpeg-xl/-/issues/51) from the reference implementation has some more info.